### PR TITLE
posts can now be deleted. addresses #28

### PR DESCRIPTION
**Pull request by nynevi**
addresses #28

created /posts page to list user posts for organizational purposes:
- users can now delete their posts using this page. along with the deletion, they will now have the ability to edit their post using this page (this eliminates the user from going to the post page)
- added "write your first post, create draft" button if no posts are available
- added a sweetalert prompt for double-conf for deletion.
- added drafts section to the /posts page with the tooltip "This section is still under development." so it can be easily integrated.
- added Load More button at the bottom which loads 20 posts on each click and adds them to the bottom. the button is only visible if user has more posts than 20 and it is hidden after user has no more posts to load.

### CODE REVIEW

File: src/core/services/PostService.ts [hunk 1]
```diff
@@ -55,6 +55,10 @@ export default class PostService {
     return this.processPost(res.data);
   }
 
+  public async deletePost(postId: number): Promise {
```
Comment (adrianbarwicki) [inline src/core/services/PostService.ts]: Please specify the type of the Promise in a way Promise<T>

File: src/app/controllers/PostsCtrl.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import tippy from "tippy.js";
+import 'tippy.js/dist/tippy.css';
+import swal from "sweetalert";
+
+import PostService from "../../core/services/PostService";
```
Comment (adrianbarwicki) [inline src/app/controllers/PostsCtrl.ts]: Not sorted alphabetically. Imports need to be sorted alphabetically. After merge of master into your branch you should get the lintner configuration with error highlighting in vs.

File: src/app/controllers/PostsCtrl.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  public loadMore() {
+    if (!this.$rootScope.activeCalls && this.postsAvailable) {
+      this.page += 1;
+      this.isLoadingMore = true;
+      this.fetchFeeds();
+    }
+  }
+
+  public fetchFeeds() {
```
Comment (adrianbarwicki) [inline src/app/controllers/PostsCtrl.ts]: Name of the method is wrong, it does not fetch feeds, but posts.

File: src/app/controllers/PostsCtrl.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    this.$timeout(() => {
+      tippy('.hc-tooltip');
+    });
+  }
+
+  private displayFeedBody(html: string): string {
+    return this.postService.displayHTML(html);
+  }
+
+  static $inject = ["$rootScope", "$scope", "$timeout", "PostService", "ScopeService"];
```
Comment (adrianbarwicki) [inline src/app/controllers/PostsCtrl.ts]: static should go as first and have "public" type. Linting from master.

**Comment by nynevi** [comment]
- made changes of the reviewer

File: src/core/models/models.ts [hunk 5]
```diff
@@ -1,5 +1,14 @@
 export type TPostTypeId = "comment" | "article" | "response";
 
+export interface IHTTPResponse {
```
Comment (adrianbarwicki) [inline src/core/models/models.ts]: This is already covered by the type: ng.IHTTpResponse

**Pull request merged**